wellsfargoAPI.py

```diff
@@ -0,0 +1,450 @@
+                sleep(2)
+                # idk why doing it through selenium doesnt work sometimes
+                driver.execute_script('document.getElementById("BuySellBtn").click()')
+                # Buy or Sell
+                if orderObj.get_action().lower() == "buy":
+                    action = WebDriverWait(driver, 20).until(
+                        EC.element_to_be_clickable((By.LINK_TEXT, "Buy"))
+                    )
+                elif orderObj.get_action().lower() == "sell":
+                    action = WebDriverWait(driver, 20).until(
+                        EC.element_to_be_clickable((By.LINK_TEXT, "Sell"))
+                    )
+                else:
+                    print("no buy or sell set")
+                action.click()
+
+                review = WebDriverWait(driver, 20).until(
+                    EC.element_to_be_clickable((By.ID, "actionbtnContinue"))
+                )
+                driver.execute_script("arguments[0].scrollIntoView(true);", review)
+                sleep(2)
+                tickerBox = WebDriverWait(driver, 20).until(
+                    EC.element_to_be_clickable((By.ID, "Symbol"))
+                )
+
+                tickerBox.send_keys(s)
+                tickerBox.send_keys(Keys.ENTER)
+
+                # quantity
+                driver.execute_script(
+                    "document.querySelector('#OrderQuantity').value ="
+                    + str(int(orderObj.get_amount()))
+                )
+
+                # get price
+                WebDriverWait(driver, 20).until(
+                    EC.presence_of_element_located((By.CLASS_NAME, "qeval"))
+                )
+
+                price = driver.find_element(By.CLASS_NAME, "qeval").text
+                price = float(price)
+                if orderObj.get_action().lower() == "buy" and price < 2:
+                    price_type = "Limit"
+                    price += 0.01
+                elif orderObj.get_action().lower() == "sell" and price < 2:
+                    price_type = "Limit"
+                    price -= 0.01
+                else:
+                    price_type = "Market"
+
+                # order type
+                driver.execute_script(
+                    "document.getElementById('OrderTypeBtnText').click()"
+                )
+
+                # limit price
+                order = driver.find_element(By.LINK_TEXT, price_type)
+                order.click()
+                if price_type == "Limit":
+                    tickerBox = driver.find_element(By.ID, "Price")
+                    tickerBox.send_keys(price)
+                    tickerBox.send_keys(Keys.ENTER)
+
+                    # timing
+                    driver.execute_script("document.getElementById('TIFBtn').click()")
+                    sleep(1)
+                    day = driver.find_element(By.LINK_TEXT, "Day")
+                    day.click()
+
+                # preview
+                review.click()
```

Comment:
This is throwing an error for me. Also there is no catching of errors in this entire code block, so Discord has no idea what is going on here:


```
C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Scripts\python.exe C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\autoRSA.py discord 
Python version: 3.12.5 (tags/v3.12.5:ff3bc82, Aug  6 2024, 20:45:27) [MSC v.1940 64 bit (AMD64)]

Starting Git auto update...
UPDATE ERROR: Conflicting changes found. Please commit, stash, or remove your changes before updating.
Using commit be5dbd5
Error running ls-remote: Branch fork/PZES/wellsfargo not found in remote origin. Perhaps you are on a fork?

Checking Python pip package versions...
[2024-09-27 13:29:03] [INFO    ] discord.client: logging in using static token
All required packages are installed and up to date.

Running Discord bot from command line

Discord bot is started...

[2024-09-27 13:29:04] [INFO    ] discord.gateway: Shard ID None has connected to Gateway (Session ID: 55eb4e84de7de717443430cd4e5f2184).
Logging into WELLS FARGO...
=====================================================


Wells Fargo Error: Message: stale element reference: stale element not found in the current frame
  (Session info: chrome=129.0.6668.70); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465922BE]
	(No symbol) [0x00007FF746594637]
	(No symbol) [0x00007FF7465946F0]
	(No symbol) [0x00007FF7465D9BC0]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]

Traceback (most recent call last):
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\wellsfargoAPI.py", line 152, in wellsfargo_holdings
    more = WebDriverWait(driver, 20).until(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\support\wait.py", line 96, in until
    value = method(self._driver)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\support\expected_conditions.py", line 365, in _predicate
    if element and element.is_enabled():
                   ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 192, in is_enabled
    return self._execute(Command.IS_ELEMENT_ENABLED)["value"]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 395, in _execute
    return self._parent.execute(command, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 354, in execute
    self.error_handler.check_response(response)
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame
  (Session info: chrome=129.0.6668.70); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465922BE]
	(No symbol) [0x00007FF746594637]
	(No symbol) [0x00007FF7465946F0]
	(No symbol) [0x00007FF7465D9BC0]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]


==============================
WELLSFARGO Holdings
==============================
WELLSFARGO 1 (0822)
No holdings in Account
Total: $0.00

WELLSFARGO 1 (8440)
No holdings in Account
Total: $0.00

==============================
Killing driver for WELLSFARGO 1
Killed 1 WELLSFARGO drivers

All commands complete in all brokers
Logging into WELLS FARGO...
=====================================================



==============================
WELLS FARGO
==============================

Traceback (most recent call last):
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\autoRSA.py", line 187, in fun_run
    globals()[fun_name](
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\wellsfargoAPI.py", line 413, in wellsfargo_transaction
    review.click()
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 94, in click
    self._execute(Command.CLICK_ELEMENT)
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 395, in _execute
    return self._parent.execute(command, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 354, in execute
    self.error_handler.check_response(response)
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element is not clickable at point (522, 840)
  (Session info: chrome=129.0.6668.70)
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465E74EE]
	(No symbol) [0x00007FF7465E4F3C]
	(No symbol) [0x00007FF7465E2408]
	(No symbol) [0x00007FF7465E161A]
	(No symbol) [0x00007FF7465D36BE]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]


Error in wellsfargo_transaction with wellsfargo: Message: element click intercepted: Element is not clickable at point (522, 840)
  (Session info: chrome=129.0.6668.70)
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465E74EE]
	(No symbol) [0x00007FF7465E4F3C]
	(No symbol) [0x00007FF7465E2408]
	(No symbol) [0x00007FF7465E161A]
	(No symbol) [0x00007FF7465D36BE]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]

Self: 
                 Action: buy
                 Amount: 1.0
                 Stock: ['ADTX', 'LSB']
                 Time: day
                 Price: market
                 Brokers: ['wellsfargo']
                 Not Brokers: []
                 Dry: True
                 Holdings: False
                 Logged In: {'wellsfargo': <helperAPI.Brokerage object at 0x000002202E6A71A0>}

All commands complete in all brokers
Logging into WELLS FARGO...
=====================================================



==============================
WELLS FARGO
==============================

Traceback (most recent call last):
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\autoRSA.py", line 187, in fun_run
    globals()[fun_name](
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\wellsfargoAPI.py", line 413, in wellsfargo_transaction
    review.click()
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 94, in click
    self._execute(Command.CLICK_ELEMENT)
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 395, in _execute
    return self._parent.execute(command, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 354, in execute
    self.error_handler.check_response(response)
  File "C:\Users\Matthew\Desktop\coding-stuff\python\auto-rsa-upstream\auto-rsa\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element is not clickable at point (522, 840)
  (Session info: chrome=129.0.6668.70)
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465E74EE]
	(No symbol) [0x00007FF7465E4F3C]
	(No symbol) [0x00007FF7465E2408]
	(No symbol) [0x00007FF7465E161A]
	(No symbol) [0x00007FF7465D36BE]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]


Error in wellsfargo_transaction with wellsfargo: Message: element click intercepted: Element is not clickable at point (522, 840)
  (Session info: chrome=129.0.6668.70)
Stacktrace:
	GetHandleVerifier [0x00007FF74675B125+29573]
	(No symbol) [0x00007FF7466CFF50]
	(No symbol) [0x00007FF74658B6EA]
	(No symbol) [0x00007FF7465E74EE]
	(No symbol) [0x00007FF7465E4F3C]
	(No symbol) [0x00007FF7465E2408]
	(No symbol) [0x00007FF7465E161A]
	(No symbol) [0x00007FF7465D36BE]
	(No symbol) [0x00007FF7466072FA]
	(No symbol) [0x00007FF7465D2FF6]
	(No symbol) [0x00007FF746607510]
	(No symbol) [0x00007FF7466286BC]
	(No symbol) [0x00007FF7466070A3]
	(No symbol) [0x00007FF7465D12DF]
	(No symbol) [0x00007FF7465D2441]
	GetHandleVerifier [0x00007FF746A8C76D+3377613]
	GetHandleVerifier [0x00007FF746AD7B67+3685831]
	GetHandleVerifier [0x00007FF746ACCF8B+3641835]
	GetHandleVerifier [0x00007FF74681B2A6+816390]
	(No symbol) [0x00007FF7466DB25F]
	(No symbol) [0x00007FF7466D7084]
	(No symbol) [0x00007FF7466D7220]
	(No symbol) [0x00007FF7466C607F]
	BaseThreadInitThunk [0x00007FF86BED257D+29]
	RtlUserThreadStart [0x00007FF86CDAAF28+40]

Self: 
                 Action: sell
                 Amount: 1.0
                 Stock: ['AAPL']
                 Time: day
                 Price: market
                 Brokers: ['wellsfargo']
                 Not Brokers: []
                 Dry: True
                 Holdings: False
                 Logged In: {'wellsfargo': <helperAPI.Brokerage object at 0x00000220303A0560>}

All commands complete in all brokers
```

Reply:
- if it is possible on this it would be helpful to see some screenshots. It looks like something is getting out of sequence and a pop-up may not be getting cleared.

Reply:
![image](https://github.com/user-attachments/assets/bb9660b5-71ec-427a-a56f-65543ac5f3d5)
@MaxxRK Sure thing, the preview button is not in sight as seen here. We could either scroll or just simply use `driver.execute_script("arguments[0].click();", review)` to fix it.